src/pybind/mgr/mirroring/fs/utils.py

```diff
@@ -0,0 +1,165 @@
+import errno
+import logging
+import threading
+from contextlib import contextmanager
+from typing import Dict
+
+import rados
+import cephfs
+
+from .exception import MirrorException
+
+MIRROR_OBJECT_PREFIX = 'cephfs_mirror'
+MIRROR_OBJECT_NAME = MIRROR_OBJECT_PREFIX
+
+INSTANCE_ID_PREFIX = "instance_"
+DIRECTORY_MAP_PREFIX = "dir_map_"
+
+log = logging.getLogger(__name__)
+
+@contextmanager
+def open_ioctx(cluster, pool_id):
+    """
+    """
+    with cluster.open_ioctx2(pool_id) as ioctx:
+        yield ioctx
+
+def connect_to_cluster(client_name, cluster_name, desc=''):
+    try:
+        log.debug(f'connecting to {desc} cluster: {client_name}/{cluster_name}')
+        r_rados = rados.Rados(rados_id=client_name, clustername=cluster_name)
+        r_rados.conf_read_file()
+        r_rados.connect()
+        log.debug(f'connected to {desc} cluster')
+        return r_rados
+    except rados.Error as e:
+        if e.errno == errno.ENOENT:
+            raise MirrorException(-e.errno, f'cluster {cluster_name} does not exist')
+        else:
+            log.error(f'error connecting to cluster: {e}')
+            raise Exception(-e.errno)
+
+def disconnect_from_cluster(cluster_name, cluster):
+    try:
+        log.debug(f'disconnecting from {cluster_name}')
+        cluster.shutdown()
+        log.debug(f'disconnected from {cluster_name}')
+    except Exception as e:
+        log.error(f'error disconnecting: {e}')
+
+def connect_to_filesystem(client_name, cluster_name, fs_name, desc):
+    try:
+        cluster = connect_to_cluster(client_name, cluster_name, desc)
+        log.debug(f'connecting to {desc} filesystem: {fs_name}')
+        fs = cephfs.LibCephFS(rados_inst=cluster)
+        log.debug('CephFS initializing...')
+        fs.init()
+        log.debug('CephFS mounting...')
+        fs.mount(filesystem_name=fs_name.encode('utf-8'))
+        log.debug(f'Connection to cephfs {fs_name} complete')
+        return (cluster, fs)
+    except cephfs.Error as e:
+        if e.errno == errno.ENOENT:
+            raise MirrorException(-e.errno, f'filesystem {fs_name} does not exist')
+        else:
+            log.error(f'error connecting to {fs_name}: {e}')
+            raise Exception(-e.errno)
+
+def disconnect_from_filesystem(cluster_name, fs_name, cluster, fs_handle):
+    try:
+        log.debug(f'disconnecting from {fs_name}')
+        fs_handle.shutdown()
+        log.debug(f'disconnected from {fs_name}')
+        disconnect_from_cluster(cluster_name, cluster)
+    except Exception as e:
+        log.error(f'error disconnecting: {e}')
```

Comment:
log message dumped during an exception in `disconnect_from_filesystem()` and `disconnect_from_cluster()` has the same message: `error disconnecting`

It would help if the message were `disconnecting from cluster` and `disconnecting from file-system` to disambiguate the messages
Is there any macro in Python that dumps the function name (like `__func__` in C++) as part of a log message ?